### PR TITLE
MONGOID-5161 revise dots and dollars documentation again


### PR DESCRIPTION
**Pull request by p-mongo**
https://jira.mongodb.com/browse/MONGOID-5161

### CODE REVIEW

File: docs/reference/fields.txt [hunk 1]
```diff
@@ -1320,41 +1320,25 @@ you can include the short versions of the modules.
...
-have been inserted using other mechanisms by field names containing dots and
-dollars, there is presently no possibility to insert such documents using
-either Ruby driver or Mongoid.
-
-To summarize, field names containing dots and dollars have limited support in
-Mongoid, the Ruby MongoDB driver and MongoDB server. Such field names should
-only be used when interfacing with software that already uses these names,
-and should be avoided for new development.
+It is recommended to not use dots/periods (``.``) in field names, and to not
+start field names with the dollar sign (``$``).
```
Comment (Neilshweky) [inline docs/reference/fields.txt:1324]: Not a big deal but I might word it:

Using dots/periods (\`\`.\`\`) in fields names and starting a field name with a dollar sign (\`\`$\`\`) is not recommended, as Mongoid provides limited support for retrieving and operating on the documents stored in those fields.

Comment (p-mongo) [reply docs/reference/fields.txt:1324]: :+1:

**Comment by johnnyshields** [comment]
This is very good and clear, thank you. I'll close my PR.

**Review by Neilshweky — APPROVED** [review]
LGTM!

**Pull request merged**